From be1020c507d56eb2d1c03a92d7c48a751b52af1d Mon Sep 17 00:00:00 2001
From: Nicola Ferralis <feranick@hotmail.com>
Date: Mon, 4 Aug 2025 15:44:21 -0400
Subject: [PATCH 5/5] Do not use noop_llseek instead of no_llseek

They are not completely equivalent. Common practice is to drop no-llseek.
See:
https://github.com/torvalds/linux/commit/cb787f4ac0c2e439ea8d7e6387b925f74576bdf8
https://github.com/torvalds/linux/commit/868941b14441282ba08761b770fc6cad69d5bdb7
---
 debian/changelog  | 4 ++--
 src/gasket_core.c | 1 -
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/debian/changelog b/debian/changelog
index a1ff0b3..3032563 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,9 +1,9 @@
-gasket-dkms (1.0-18.3) noble; urgency=medium
+gasket-dkms (1.0-18.4) noble; urgency=medium
  
   * Update ffor Kernel 6.13+
     - Remove deprecated call REMAKE_INITRD.
     - Use literal for MODULE_IMPORT_NS.
-    - Use "noop_llseek" instead of deprecated "no_llseek".  
+    - Remove deprecated "no_llseek".  
 
  -- Nicola Ferralis <feranick@hotmail.com>  Mon, 04 Aug 2025 14:41:16 -0400
 
diff --git a/src/gasket_core.c b/src/gasket_core.c
index 4e5ad8c..febaade 100644
--- a/src/gasket_core.c
+++ b/src/gasket_core.c
@@ -1373,7 +1373,6 @@ static long gasket_ioctl(struct file *filp, uint cmd, ulong arg)
 /* File operations for all Gasket devices. */
 static const struct file_operations gasket_file_ops = {
 	.owner = THIS_MODULE,
-	.llseek = noop_llseek,
 	.mmap = gasket_mmap,
 	.open = gasket_open,
 	.release = gasket_release,
-- 
2.51.0

